account_test.go

```diff
@@ -57,13 +154,97 @@ func TestAccountUnmarshal(t *testing.T) {
 	err = json.Unmarshal(bytes, &account)
 	assert.NoError(t, err)
 
-	assert.Equal(t, AccountTypeCustom, account.Type)
 	assert.Equal(t, "acct_123", account.ID)
+
+	assert.Equal(t, "value1", account.Metadata["key1"])
+	assert.Equal(t, "value2", account.Metadata["key2"])
+
+	assert.Equal(t, int64(2), account.PayoutSchedule.DelayDays)
+	assert.Equal(t, PayoutIntervalWeekly, account.PayoutSchedule.Interval)
+
+	assert.Equal(t, AccountTypeCustom, account.Type)
+
+	assert.Equal(t, int64(1528573382), account.TOSAcceptance.Date)
+	assert.Equal(t, "127.0.0.1", account.TOSAcceptance.IP)
+	assert.Equal(t, "user agent", account.TOSAcceptance.UserAgent)
+
+	assert.Equal(t, IdentityVerificationDisabledReasonFieldsNeeded, account.Verification.DisabledReason)
+	assert.Equal(t, int64(1528573382), account.Verification.DueBy)
+	assert.Equal(t, 2, len(account.Verification.FieldsNeeded))
+	assert.Equal(t, "legal_entity.verification.document", account.Verification.FieldsNeeded[0])
+	assert.Equal(t, "legal_entity.business_name", account.Verification.FieldsNeeded[1])
+
+	// Assert ExternalAccounts are fully deserialized
 	assert.Equal(t, true, account.ExternalAccounts.HasMore)
 
 	assert.Equal(t, 2, len(account.ExternalAccounts.Data))
 	assert.Equal(t, "ba_123", account.ExternalAccounts.Data[0].ID)
 	assert.Equal(t, "card_123", account.ExternalAccounts.Data[1].ID)
+
+	// Ensure LegalEntity is fully deserialized
```

Comment:
I made thorough tests of almost every property, not sure if too much or not, please let me know if I should cut those in half.

Reply:
Probably not necessary, but I think it's okay now that you've already written it.

Reply:
Out of curiosity what do you think would have been a good test? I think this large structure is fairly easy to mess up as there are so many edge-cases. You could end up deserializing the wrong property in the wrong place like maiden name as first_name or vice versa, or identification documents like I did.

Reply:
Probably one check per "level" (i.e., level of structures which you descend to) is probably a good idea. Agree that it's easy to misname a JSON field, but I think to some degree you need to rely on the fact that these are likely to trend towards correctness over time. Adding tests gives you a little more guarantee, but the tests themselves are likely to be butchered during any sizable refactor, and so not end up testing for regressions as well as you might hope anyway.

Reply:
Gotcha! I see where you're coming from. I think I had similar discussions with @ob-stripe on the stripe-java refactor so it is starting to make sense to me.

Reply:
Yeah, it's definitely a little subjective :$